### PR TITLE
Use SourceParser on PKLookup

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Passes tests even before we write oids to source since by default References have OID_UNASSIGNED and they fall back to take name "as is" behavior.

Applied this in [#14662](https://github.com/crate/crate/pull/14662) where oid is actually written. Tests where PKLookup is happening (including `UpdateToInsert`), pass with this PR applied and fail without.

### CODE REVIEW

**Comment by BaurzhanSakhariev** [comment]
retest this please 
Jenkins hangs on assigning a machine

**Comment by BaurzhanSakhariev** [comment]
Hi @seut, could you please take a look?

This is similar to `SourceLookup` fixing, reuses already introduced `DocTableInfo.lookupNameBySourceKey()` to rewrite keys.

I had [another approach](https://github.com/crate/crate/pull/14662/commits/4f3120af2d07d9b6b38e995fdd81384941ac5256) with rewriting keys in 
`DocRefResolver` but faced an issue with passing through a Function `DocTableInfo.lookupNameBySourceKey()`.

 `PKLookupPhase` is Streamable and had to figure out how to serialize the function which was essentially pass needed data and resolve function + [weird stuff](https://github.com/crate/crate/pull/14662/commits/27be10beb2bf4dac588b0b64c5361dffc9142076#diff-dde23d148cae23c6a02cbc95790ea31ed2db14f8033958a4e3e47c21159d54efR88-R90) to avoid NPE. 
 
Basically, abandoned approach was doing late transformation and this PR does early transformation.

**Comment by BaurzhanSakhariev** [comment]
More general idea - `PKLookup` could just reuse `SourceParser` which is already OID aware on the branch.

We can not only have less similar code but also don't hit such scenarios where we need to apply a fix to both components.

Recent example:
We modified `SourceParser` to handle empty arrays in https://github.com/crate/crate/pull/14049
And then had to follow up in https://github.com/crate/crate/pull/14255 to deal with PK scenario.

With the single entry point we could have modified only once.

P.S Wild shot: somehow make translog read operation re-use `SourceParser`, ie `applyTranslogOperation` should not create `SourceToParse` but pass source as is and parse it with our component instead.
